src/matrix.js

```diff
@@ -82,21 +179,47 @@ SVG.Matrix = SVG.invent({
 
       return matrix
     },
-    // Multiplies by given matrix
+
+    // Left multiplies by the given matrix
     multiply: function (matrix) {
-      return new SVG.Matrix(this.native().multiply(parseMatrix(matrix).native()))
+      // Get the matrices
+      var l = this
+      var r = parseMatrix(matrix)
+
+      // Work out the product directly
+      var a = l.a * r.a + l.c * r.b
+      var b = l.b * r.a + l.d * r.b
+      var c = l.a * r.c + l.c * r.d
+      var d = l.b * r.c + l.d * r.d
+      var e = l.e + l.a * r.e + l.c * r.f
+      var f = l.f + l.b * r.e + l.d * r.f
+
+      // Form the matrix and return it
+      var product = new SVG.Matrix(a, b, c, d, e, f)
+      return product
     },
+
+    lmultiply: function (matrix) {
+      var l = parseMatrix(matrix)
+      return l.multiply(this)
+    },
```

Comment:
you could go `return parseMatrix(matrix).multiply(this)` directly if you like.
Also: What is parseMatrix doing? Shouldnt be a `new SVG.Matrix(matrix)` be sufficiant?

Reply:
I'm not sure. I just kept parseMatrix because it was already there. I didn't add it? What is parseMatrix doing :P I think it also allows for strings. Does the method you mentioned work with strings?

Reply:
yes it can handle strings. parseMatrix actually does nothing:

```js
// Parse matrix if required
function parseMatrix (matrix) {
  if (!(matrix instanceof SVG.Matrix)) {
    matrix = new SVG.Matrix(matrix)
  }

  return matrix
}
```

Ok maybe its better. It does not clones teh matrix. its a bit faster. so maybe we should always use that one :D

Reply:
Done!